OpenRA.Game/Map/Map.cs

```diff
@@ -285,6 +287,26 @@ public static string ComputeUID(IReadOnlyPackage package)
 			}
 		}
 
+		// PERF This is a way to get MapFormat without expensive yaml parsing
+		public static int GetMapFormat(IReadOnlyPackage p)
+		{
+			var result = "";
+			foreach (var line in p.GetStream("map.yaml").ReadAllLines())
+			{
+				var search = Regex.Match(line, "MapFormat: (.?\\d)");
+				if (search.Success && search.Groups.Count > 0)
+				{
+					result = search.Groups[1].Value;
```

Comment:
Change this to `﻿return FieldLoader.GetValue<int>("MapFormat", search.Groups[1].Value);` then you can remove the `break`, the braces, and throw unconditionally after the loop.

Reply:
ok